shared/pkg/scanner/grype/local_grype.go

```diff
@@ -68,13 +68,20 @@ func (s *LocalScanner) run(sourceType utils.SourceType, userInput string) {
 		return
 	}
 
+	var hash string
+	origInput := userInput
 	if sourceType == utils.SBOM {
 		syftJSONFilePath, cleanup, err := ConvertCycloneDXFileToSyftJSONFile(userInput, s.logger)
 		if err != nil {
 			ReportError(s.resultChan, fmt.Errorf("failed to convert sbom file: %w", err), s.logger)
 			return
 		}
 		defer cleanup()
+		origInput, hash, err = getOriginalInputAndHashFromSBOM(userInput)
```

Comment:
do we want to validate that the hash is not empty even if there was no error?

Reply:
In the `getSource()` we check the hash is empty or not. If the hash is empty string wi will try to get it from the repodigest or manifestdigest.
https://github.com/openclarity/kubeclarity/blob/f145970f20686b1cde40f5d4eb62991b68b83dae/shared/pkg/scanner/grype/common.go#L133-L146

Reply:
so we assume that ```manifestHash := doc.Source.Target.(syft_source.ImageMetadata).ManifestDigest``` will always get something?

Reply:
If everything is empty we don't have any hash and we can't calculate it as well because we are using sbom as input. Se we let the hash empty.

Reply:
The hash can be empty in that case when we are using third party sbom that doesn't care about the hash.